tests/PolyType.Tests/TypeShapeProviderTests.cs

```diff
@@ -362,6 +362,13 @@ public void GetDictionaryType<T>(TestCase<T> testCase)
         }
     }
 
+    [Fact]
+    public void IsTagSpecified()
+    {
+        Assert.True(((IUnionTypeShape?)Provider.GetShape(typeof(Tree)))?.UnionCases[0].IsTagSpecified);
```

Comment:
Could you convert this to a theory that validates the value of the property based on the result from querying the relevant attribute?

Reply:
Actually, there's a theory on this class already that specifically checks `IUnionTypeShape` instances. Incorporating it there could work as well.